examples/rigid/debug.py

```diff
@@ -0,0 +1,434 @@
+                pos=new_pos,
+                size=new_size,
+                radius=new_radius,
+                height=new_height,
+            )
+            dict_obj_idx[obj_idx] = [added_obj, obj_shape]
+
+    # set camera pos to the highest place
+    max_z = 0
+    max_x = 0
+    max_y = 0
+    for meta in list_occupied:
+        new_pos, new_size, new_radius, new_height, new_shape = meta
+        if new_shape == "Box":
+            max_x = max(max_x, new_pos[0] + new_size[0] / 2)
+            max_y = max(max_y, new_pos[1] + new_size[1] / 2)
+            max_z = max(max_z, new_pos[2] + new_size[2] / 2)
+        elif new_shape == "Sphere":
+            max_x = max(max_x, new_pos[0] + new_radius)
+            max_y = max(max_y, new_pos[1] + new_radius)
+            max_z = max(max_z, new_pos[2] + new_radius)
+        elif new_shape == "Cylinder":
+            max_x = max(max_x, new_pos[0] + new_radius)
+            max_y = max(max_y, new_pos[1] + new_radius)
+            max_z = max(max_z, new_pos[2] + new_height / 2)
+
+    max_x = np.ceil(max_x + 20)
+    max_y = np.ceil(max_y + 20)
+    max_z = np.ceil(max_z + 3)
+    cam = scene.add_camera(
+        res=(640, 480),
+        pos=(max_x, max_y, max_z),
+        lookat=(0.0, 0.0, 0.0),
+        fov=35,
+        GUI=False,
+    )
+
+    scene.build(n_envs=B, env_spacing=(1.0, 1.0))
+
+    for obj_idx in range(num_obj):
+        added_obj = dict_obj_idx[obj_idx][0]
+        added_obj.set_friction_ratio(
+            friction_ratio=0.5 + torch.rand(scene.n_envs, added_obj.n_links),
+            ls_idx_local=np.arange(0, added_obj.n_links),
+        )
+
+    for solver in scene.sim.solvers:
+        if not isinstance(solver, RigidSolver):
+            continue
+        rigid_solver = solver
+
+    # cam.start_recording()
+    horizon = 500
+    recorded_depth = []
+    recorded_seg = []
+    recorded_seg_id = []
+
+    # number of entity
+    num_entities = len(scene.rigid_solver.entities)
+    # create segments colors
+    seg_idxc_to_color = np.random.randint(0, 255, (num_entities, 3), dtype=np.uint8)
+    # set background colors
+    seg_idxc_to_color[0, :] = 0
+    link_idx = list(range(1, num_entities))
+    list_changes = []
+    for i in link_idx:
+        obj_shape = dict_obj_idx[i - 1][1]
+        if obj_shape not in ("Sphere", "Cylinder"):
+            action = np.random.choice(["force", "torque"])
+        else:
+            action = "force"
+
+        if action == "force":
+            # f_scale = np.random.rand() * 50 - 25
+            f_scale = np.random.rand() * 50 + 20
+            list_changes.append(["force", i, f_scale])
+        else:
+            # rot_scale = np.random.rand() * 20 - 10
+            rot_scale = np.random.rand() * 20 + 20
+            list_changes.append(["torque", i, rot_scale])
+
+    os.makedirs(output_folder, exist_ok=True)
+    rotation_direction = 1
+
+    for i in range(horizon):
+
+        if i == 0:
+            # for obj_id in dict_obj_idx.keys():
```

Comment:
Avoid comments in the code unless they are meant to be uncommented/commented by the end-user.